pkg/engine/armresources.go

```diff
@@ -22,9 +22,15 @@ func GenerateARMResources(cs *api.ContainerService) []interface{} {
 		userAssignedIDEnabled = useManagedIdentity && kubernetesConfig.UserAssignedID != ""
 	}
 
+	isHostedMaster := cs.Properties.IsHostedMasterProfile()
 	if userAssignedIDEnabled {
 		userAssignedID := createUserAssignedIdentities()
-		msiRoleAssignment := createMSIRoleAssignment()
+		var msiRoleAssignment RoleAssignmentARM
+		if isHostedMaster {
+			msiRoleAssignment = createMSIRoleAssignment(IdentityReaderRole)
```

Comment:
@jackfrancis @CecileRobertMichon Shall we change this `ReaderRole` to `ContributorRole` on agent node? Since we are going to support CSI driver with MSI credential. And CSI driver will be installed on agent nodes, e.g. disk provisioner of azure disk CSI driver will be installed as a StatefulsSet on one agent node, `ReaderRole` MSI won't work since it needs write disk permission.

Reply:
If I understand this correctly, CSI should be an option, something like `cs.Properties.****.UseCSI`, 
and its default value is `false`, so I personally prefer to add a judgment somewhere: if CSI is used, use `Contributor` role on agent node; otherwise, use `Reader` role.

Reply:
In the future, it's not an option, it's a must, I think we need to enable it now.
CSI is beta now, would step into GA soon.

Reply:
Got that. This PR only affects cluster having hosted master. For a complete cluster using user assigned identity, the identity still has `Contributor` role, so it won't affect CSI's usage. When CSI becomes mandatory in AKS, we can revert this PR and grant the user assigned identity `Contributor` role. Does that make sense?

Reply:
@norshtein then why not enabled it now? AKS v1.13.x is already available now, if we enable `Contributor` role, then we could test CSI driver on AKS at the first moment when MSI is enabled on AKS.

Reply:
What's the status of CSI driver on AKS? Are users able to enable(and disable) CSI?